### PR TITLE
Adds "Reset" button to refine options

### PR DESCRIPTION
**Pull request by ashfurrow**
Related to https://github.com/artsy/eigen/issues/971

### CODE REVIEW

File: Artsy/View_Controllers/App_Navigation/ARTopMenuViewController+DeveloperExtras.m [hunk 1]
```diff
@@ -12,6 +12,13 @@ @implementation ARTopMenuViewController (DeveloperExtras)
 
 - (void)runDeveloperExtras
 {
+    static dispatch_once_t onceToken;
```
Comment (orta) [inline Artsy/View_Controllers/App_Navigation/ARTopMenuViewController+DeveloperExtras.m]: this shouldn't be in a PR

File: Artsy_Tests/Extensions/UIViewController+Tests.swift [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import UIKit
+
+extension UIViewController {
+    /// Loads the view by faking a non-animated (dis)appearance transition.
+    func loadViewProgrammatically(appearing: Bool = true) {
+        beginAppearanceTransition(appearing, animated: false)
+        endAppearanceTransition()
+    }
+}
```
Comment (orta) [inline Artsy_Tests/Extensions/UIViewController+Tests.swift]: if you're introducing this, there are a few places where we do both in tests that should change to this API :+1:

File: Artsy/View_Controllers/App_Navigation/ARTopMenuViewController+DeveloperExtras.m (hunk 1 above)
Comment (orta) [inline Artsy/View_Controllers/App_Navigation/ARTopMenuViewController+DeveloperExtras.m]: I've added this to danger on #1050

Comment (ashfurrow) [inline Artsy/View_Controllers/App_Navigation/ARTopMenuViewController+DeveloperExtras.m]: Ah sorry, had sone trouble with a rebase and this slipped in :grimacing:

File: Artsy_Tests/Extensions/UIViewController+Tests.swift (hunk 2 above)
Comment (ashfurrow) [inline Artsy_Tests/Extensions/UIViewController+Tests.swift]: Cool, I'll take a look.

Comment (ashfurrow) [inline Artsy_Tests/Extensions/UIViewController+Tests.swift]: I've opened an issue for this: https://github.com/artsy/eigen/issues/1074

**Comment by ArtsyMobile** [comment]
&nbsp; | 4 Messages
    ------------- | ------------
        :book: | Insertions: 0
        :book: | Added: []
        :book: | LOC: 7
        :book: | Mod: ["Artsy/View_Controllers/App_Navigation/ARTopMenuViewController+DeveloperExtras.m"]
    

  


<p align="right" meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

**Comment by ashfurrow** [comment]
@orta are you OK with merging without applying `loadViewProgrammatically` to existing tests?

File: Artsy/View_Controllers/Auction/AuctionRefineViewController+Private.swift [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+private let CellIdentifier = "Cell"
+
+extension AuctionRefineViewController {
+    func setupViews() {
+        let cancelButton = self.cancelButton()
+        view.addSubview(cancelButton)
+        cancelButton.alignTopEdgeWithView(view, predicate: "10")
+        cancelButton.alignTrailingEdgeWithView(view, predicate: "-10")
+
+        let titleLabel = ARSerifLabel().then {
```
Comment (alloy) [inline Artsy/View_Controllers/Auction/AuctionRefineViewController+Private.swift]: Is the `.then` part base Swift? Is it just sugar to yield an object and return it again?

Comment (ashfurrow) [inline Artsy/View_Controllers/Auction/AuctionRefineViewController+Private.swift]: It's a pod called Then, actually. Does exactly what you think.

--

[A.]           Ash Furrow

Artsy <https://www.artsy.net/>

                *ash@artsy.net <ash@artsy.net>*

                (347)-592-4474 <//(347)-592-4474> (Mobile)
                @ashfurrow <http://twitter.com/ashfurrow>

**Comment by alloy** [comment]
LGTM :+1:

File: Artsy/View_Controllers/Auction/AuctionRefineViewController+Private.swift (hunk 3 above)
Comment (alloy) [inline Artsy/View_Controllers/Auction/AuctionRefineViewController+Private.swift]: Cool!

I’m not sure I like the name vs Ruby’s `tap` (tapping into a pipeline), but that might just be me being used to it :)

http://moonbase.rydia.net/mental/blog/programming/eavesdropping-on-expressions.html

Comment (ashfurrow) [inline Artsy/View_Controllers/Auction/AuctionRefineViewController+Private.swift]: Neat! I had seen `tap` used before, the pipeline metaphor makes it make sense now :)

**Pull request merged**